packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.services.explorer;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import org.eclipse.sirius.components.collaborative.trees.api.IInitialDirectEditTreeItemLabelProvider;
+import org.eclipse.sirius.components.collaborative.trees.dto.InitialDirectEditElementLabelInput;
+import org.eclipse.sirius.components.collaborative.trees.dto.InitialDirectEditElementLabelSuccessPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.trees.Tree;
+import org.eclipse.sirius.components.trees.TreeItem;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to compute the initial label to display when the direct edit of a tree item of the explorer is triggered on the frontend.
+ *
+ * @author mcharfadi
+ */
+@Service
+public class ExplorerInitialDirectEditTreeItemLabelProvider implements IInitialDirectEditTreeItemLabelProvider {
+
+    public static final String EXPLORER_DESCRIPTION_ID = UUID.nameUUIDFromBytes("explorer_tree_description".getBytes()).toString();
+
+    public static final String EXPLORER_DOCUMENT_KIND = "siriusComponents://representation?type=Tree";
+
+    public static final String EXPLORER_NAME = "Explorer";
+
+    @Override
+    public boolean canHandle(Tree tree) {
+        return tree.getId().startsWith("explorer://");
+    }
+
+    @Override
+    public IPayload handle(IEditingContext editingContext, Tree tree, InitialDirectEditElementLabelInput input) {
+        String initialLabel = tree.getChildren().stream()
```

Comment:
I have not tested, but it's possible this might be simplifiable into something like:

```java
new TreeQueryService().findTreeItem(tree, input.treeItemId()).map(TreeItem::getLabel).orElse("")
```

Reply:
Yes it worked, thanks you